src/scss/components/_btn_text.scss

```diff
@@ -0,0 +1,141 @@
+
+// light-or-dark($color)
+// Use contrast against white or black to determine if a color is "light" or "dark"
+// Adapted from: https://medium.com/dev-channel/using-sass-to-automatically-pick-text-colors-4ba7645d2796
+@function light-or-dark($color) {
+  $light-contrast: contrast-ratio($color, $white);
+  $dark-contrast: contrast-ratio($color, $black);
+
+  @if $light-contrast > $dark-contrast {
+    // Contrast against white is higher than against black, so, this is a dark color
+    @return "dark";
+  } @else {
+    @return "light";
+  }
+}
+
+// a11y-color($color-1, $color-2)
+// $color-1: foreground color
+// $color-2: background color
+// Goal: Return 2 colors that pass for the chosen WCAG level without changing the Hue of the color
+// ratio: 3.0, 4.5 (= AA compliant), or 7.0
+@function a11y-color($color-1, $color-2, $lod-color-2: 'null', $ratio: 4.5) {
+  $adjust-color-1: 'false';
+  // Calculate the first contrast ratio of the given pair
+  $original-contrast: contrast-ratio($color-1, $color-2);
+
+  @if $original-contrast >= $ratio {
+    // If we pass the ratio already, return the original color
+    @return ($color-1, $color-2);
+  } @else {
+    // Doesn't pass
+    // Should the color be lightened or darkened?
+    $color-1-lod: light-or-dark($color-1);
+    $color-2-lod: light-or-dark($color-2);
+
+    // Set a "step" value to lighten or darken a color
+    // Note: Higher percentage steps means faster compile time, but we might overstep the required threshold too far with something higher than 5%
+    $step: 2%;
+
+    @if $color-2-lod == 'light' and $lod-color-2 == 'lighter' {
+      // if $color-1 is light and $lod-color-2 is lighter -> darken $color-2
+      $adjust-color-1: 'true';
+      $step: - $step;
+    }
+    @else if $color-2-lod == 'light' and $color-1-lod == 'light' {
+      // Both are light colors, darken the color-1
+      $step: - $step;
+    } @else if $color-2-lod == 'dark' and $color-1-lod == 'light' {
+      // color-1 is light, color-2 is dark but does not pass, darken more
+      $step: - $step;
+    }
+
+    // Loop through with a @while statement until the color combination passes our required ratio. Scale the color by our step value until the expression is false
+    // This might take some depending on the colors -> pay attention that the loop is not infinite!
+    @while contrast-ratio($color-1, $color-2) < $ratio {
+      @if $adjust-color-1 == 'true' {
+        $color-1: scale-color($color-1, $lightness: $step, $saturation: $step/2)
+      }
+      @else {
+        $color-2: scale-color($color-2, $lightness: $step, $saturation: $step/2);
+      }
+    }
+    @return ($color-1, $color-2);
+  }
+}
+
+
+@mixin btn-text-common() {
+
+  // Align all children vertically
+  > * {
+    vertical-align: middle;
+  }
+}
+
+
+// also add gray variants to use i.e. .btn-text-gray
+$theme-colors-and-gray: map-merge(("gray": $gray-600), $theme-colors);
```

Comment:
@dvdanielamoitzi @thinkh better add it here directly to allow for any btn-gray variant: https://github.com/phovea/phovea_ui/blob/master/src/scss/vendors/bootstrap/_1_variables_mixins_utilities.scss#L15

Reply:
very good idea, I'll move it! :+1: